### PR TITLE
octree dir: split content into smaller files

### PR DESCRIPTION
**Pull request by catevita**
ptal! @nnmm 
ptal! @feuerste 

in this small PR I splitted the node and mod files of the src/octree into pieces to allow for a better overview.

### CODE REVIEW

File: src/octree/node_iterator.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            point.color.blue = self.rgb_reader.read_u8().unwrap();
+            if let Some(ir) = self.intensity_reader.as_mut() {
+                point.intensity = Some(ir.read_f32::<LittleEndian>().unwrap());
+            }
+            f(&point);
+        }
+    }
+}
+
+fn fixpoint_decode<T>(value: T, min: f32, edge_length: f32) -> f32
```
Comment (feuerste) [inline src/octree/node_iterator.rs]: How about having `fixpoint_decode`, `decode`, `fixpoint_encode`, and `encode` in its own file, which is used in the iterator and writer?

Comment (catevita) [reply src/octree/node_iterator.rs]: done

Comment (feuerste) [reply src/octree/node_iterator.rs]: Thanks Caterina.
How about renaming the file to `codec.rs`?

Comment (catevita) [reply src/octree/node_iterator.rs]: ok! thank you for the suggestion

**Comment by catevita** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user catevita. Merge queue now has a length of 1.

**Pull request merged**